### ISSUE TITLE
Pandora ads

### ISSUE THREAD

**Issue by Gh0stWires** [02strich/python-pandora#11]
I was testing out my program to see if it truly has no limits when it comes to skipping songs. after awhile of clicking skip, ads started to play(even though its programmed to ignore ad tokens), instead of what it told me it was supposed to be playing. is there a reason why it's doing this? and is there a way around it?

**Comment by Gh0stWires** [comment]
What it seems to be doing is sending a url that is an ad where the music url should be.

**Comment by 02strich** [comment]
Limiting the number of allowed skips is one of Pandoras main points for limiting the free account, so one should be careful when circumventing that. In that context I am also hesitant to make to many guesses, but would recommend to buy a Pandora One subscription if you really aim for unlimited skipping. Sorry..

**Comment by Gh0stWires** [comment]
Then how does pithos do it?

**Comment by 02strich** [comment]
What exactly?

**Comment by Gh0stWires** [comment]
There is a program called pithos on linux. its written in python. It has no ads and unlimited skipping and never seems to run into a problem.http://pithos.github.io/

**Comment by 02strich** [comment]
I am well aware of Pithos (the code of python-pandora is originally inspired by it) and I think you got me slightly wrong: I am hessitant to support ad-free playing in the library because there were repercussion (mainy changing the API to make it deliberately complicated) in early days of Pandora against libraries that allowed this. So its not a technical thing, but more a social one.

Nevertheless having had a look at the Pithos logic, they are simply looking at the playtime:

```
if dur_int < 45.0: # Less than 45 seconds we assume it's an ad
    logging.info('Ad detected!')
    self.current_song.is_ad = True
```

You could use the same logic in your app, as this will work well for mainstream music.

**Comment by Gh0stWires** [comment]
I understand why your hesitant. I realize pandora would change there stuff if you could exploit it, because they'd loose money. The application I'm making wont be mass distributed i just made it to see if i could sort of like a technical challenge. I'm not in it to screw over pandora. also that logic from pithos is interesting i wish i thought of it.

**Comment by 02strich** [comment]
Expected so, but wanted to explain the background first :) also had the same reaction, when seeing their code...

**Comment by Gh0stWires** [comment]
yeah I understand. While this thread is open can I ask, is the add_feedback function for thumbs up...and what does add seed do?

**Comment by 02strich** [comment]
add_feedback is for thumbs up and add_seed adds a musician or similar to a station

**Comment by Gh0stWires** [comment]
Oh I didn't even know that was a feature of pandora lol
